homeassistant/components/light/tellduslive.py

```diff
@@ -0,0 +1,65 @@
+"""
+Support for Tellstick switches using Tellstick Net.
+
+This platform uses the Telldus Live online service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.tellduslive/
+
+"""
+import logging
+
+from homeassistant.components.tellduslive import (
+    TelldusLiveEntity,
+    TELLSTICK_TURNON,
+    TELLSTICK_TURNOFF,
+    TELLSTICK_DIM)
+from homeassistant.components.light import (
+    Light,
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup lights."""
+    if discovery_info is None:
+        return
+    add_devices(TelldusLiveLight(hass, light) for light in discovery_info)
+
+
+class TelldusLiveLight(TelldusLiveEntity, Light):
+    """Representation of a light."""
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        try:
+            return int(self._device.get("statevalue"))
+        except ValueError:
+            return 255
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        # pylint: disable=attribute-defined-outside-init
+        self._brightness = self.brightness  # remember last set brightness
```

Comment:
I agree that side effects is not desirable - what I wanted to achieve was to reset the dim brightness to the level before turning the light off, i.e. the scenario 1) set light to dim level 70%, 2) turn light off (equivalent to dim level 0%) 2) turn light on (and have it automatically set to 70% instead of 100%).

For that, a method was needed that was called both when the state was changed locally (i.e. by HA), or remotely (i.e. from a remote control or using the online service), and overriding `is_on` seemed like the simplest way. Will move to `update()` if more appropriate.